src/cobra/util/array.py

```diff
@@ -1,18 +1,25 @@
 """Helper functions for array operations."""
 
-from collections import namedtuple
+from typing import TYPE_CHECKING, Any, NamedTuple, Optional, Union
 
 import numpy as np
 import pandas as pd
 
 
+# Used to avoid cyclic reference and enable third-party static type checkers to work
+if TYPE_CHECKING:
+    import cobra
+
+
 try:
     from scipy.sparse import dok_matrix, lil_matrix
 except ImportError:
     dok_matrix, lil_matrix = None, None
 
 
-def create_stoichiometric_matrix(model, array_type="dense", dtype=None):
+def create_stoichiometric_matrix(
+    model: "cobra.core.Model", array_type: str = "dense", dtype: Optional[Any] = None
```

Comment:
`dtype` doesn't have a common base type?

```suggestion
    model: "Model", array_type: str = "dense", dtype: Optional[Any] = None
```

Reply:
> `dtype` doesn't have a common base type?

To maintain consistency, I think `numpy.dtype` could be used.